selfservice/strategy/link/strategy_recovery_test.go

```diff
@@ -181,8 +185,70 @@ func TestRecovery(t *testing.T) {
+		Traits:   identity.Traits(`{"email":"recoverme2@ory.sh"}`),
+		SchemaID: config.DefaultIdentityTraitsSchemaID,
+	}
+	var recoveryEmail2 = gjson.GetBytes(identityToRecover2.Traits, "email").String()
+
+	var identityToRecover3 = &identity.Identity{
+		Credentials: map[identity.CredentialsType]identity.Credentials{
+			"password": {Type: "password", Identifiers: []string{"recoverme3@ory.sh"}, Config: sqlxx.JSONRawMessage(`{"hashed_password":"foo"}`)}},
+		Traits:   identity.Traits(`{"email":"recoverme3@ory.sh"}`),
+		SchemaID: config.DefaultIdentityTraitsSchemaID,
+	}
+	var recoveryEmail3 = gjson.GetBytes(identityToRecover3.Traits, "email").String()
+
+	var identityToRecover4 = &identity.Identity{
+		Credentials: map[identity.CredentialsType]identity.Credentials{
+			"password": {Type: "password", Identifiers: []string{"recoverme4@ory.sh"}, Config: sqlxx.JSONRawMessage(`{"hashed_password":"foo"}`)}},
+		Traits:   identity.Traits(`{"email":"recoverme4@ory.sh"}`),
+		SchemaID: config.DefaultIdentityTraitsSchemaID,
+	}
+	var recoveryEmail4 = gjson.GetBytes(identityToRecover4.Traits, "email").String()
+
+	require.NoError(t, reg.IdentityManager().Create(context.Background(), identityToRecover1,
+		identity.ManagerAllowWriteProtectedTraits))
+	require.NoError(t, reg.IdentityManager().Create(context.Background(), identityToRecover2,
+		identity.ManagerAllowWriteProtectedTraits))
+	require.NoError(t, reg.IdentityManager().Create(context.Background(), identityToRecover3,
 		identity.ManagerAllowWriteProtectedTraits))
+	require.NoError(t, reg.IdentityManager().Create(context.Background(), identityToRecover4,
+		identity.ManagerAllowWriteProtectedTraits))
+
+	addr, err := reg.IdentityPool().FindVerifiableAddressByValue(context.Background(), identity.VerifiableAddressTypeEmail, recoveryEmail1)
+	assert.NoError(t, err)
+	assert.False(t, addr.Verified)
+	assert.Equal(t, sqlxx.NullTime{}, addr.VerifiedAt)
+	assert.Equal(t, identity.VerifiableAddressStatusPending, addr.Status)
+
+	addr, err = reg.IdentityPool().FindVerifiableAddressByValue(context.Background(), identity.VerifiableAddressTypeEmail, recoveryEmail2)
+	assert.NoError(t, err)
+	assert.False(t, addr.Verified)
+	assert.Equal(t, sqlxx.NullTime{}, addr.VerifiedAt)
+	assert.Equal(t, identity.VerifiableAddressStatusPending, addr.Status)
+
+	addr, err = reg.IdentityPool().FindVerifiableAddressByValue(context.Background(), identity.VerifiableAddressTypeEmail, recoveryEmail3)
+	assert.NoError(t, err)
+	assert.False(t, addr.Verified)
+	assert.Equal(t, sqlxx.NullTime{}, addr.VerifiedAt)
+	assert.Equal(t, identity.VerifiableAddressStatusPending, addr.Status)
+
+	addr, err = reg.IdentityPool().FindVerifiableAddressByValue(context.Background(), identity.VerifiableAddressTypeEmail, recoveryEmail4)
+	assert.NoError(t, err)
+	assert.False(t, addr.Verified)
+	assert.Equal(t, sqlxx.NullTime{}, addr.VerifiedAt)
+	assert.Equal(t, identity.VerifiableAddressStatusPending, addr.Status)
```

Comment:
Could you please wrap this in a function or a for loop to avoid the code duplication?

Reply:
done

Reply:
done